### PR TITLE
SNOW-1361405 Emit Span for UDx

### PR DESCRIPTION
**Pull request by sfc-gh-bli**
Emit Span for UDF, UDTF, and Stord Proc

### CODE REVIEW

File: src/test/java/com/snowflake/snowpark_test/JavaUDTFNonStoredProcSuite.java [hunk 1]
```diff
@@ -119,6 +119,8 @@ public void testPermanentUdtf() {
       Row[] rows2 = getAnotherSession().tableFunction(permFunc, lit(3), lit(4)).collect();
       assert rows2.length == 1 && rows2[0].size() == 1 && rows2[0].getInt(0) == 7;
 
+      getSession().sql("drop function " + permFunc.funcName() + "(int, int)").collect();
```
Comment (sfc-gh-yuwang) [inline src/test/java/com/snowflake/snowpark_test/JavaUDTFNonStoredProcSuite.java:122]: Is this change related to emit span for udf?

File: src/test/scala/com/snowflake/snowpark_test/UdxOpenTelemetrySuite.scala [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    super.beforeAll
+    if (!isStoredProc(session)) {
+      TestUtils.addDepsToClassPath(session)
+    }
+  }
+
+  test("udf") {
+    val className = "snow.snowpark.UDFRegistration"
+    val func = () => 100
+    session.udf.registerTemporary(func)
```
Comment (sfc-gh-yuwang) [inline src/test/scala/com/snowflake/snowpark_test/UdxOpenTelemetrySuite.scala:18]: Is this the only way to create udf in java?

File: src/test/java/com/snowflake/snowpark_test/JavaUDTFNonStoredProcSuite.java (hunk 1 above)
Comment (sfc-gh-bli) [reply src/test/java/com/snowflake/snowpark_test/JavaUDTFNonStoredProcSuite.java:122]: no, it is unrelated. I found a bug in this test, the test function hasn't been dropped after test.

File: src/test/scala/com/snowflake/snowpark_test/UdxOpenTelemetrySuite.scala (hunk 2 above)
Comment (sfc-gh-bli) [reply src/test/scala/com/snowflake/snowpark_test/UdxOpenTelemetrySuite.scala:18]: There are for ways to create UDF.
```scala
session.udf.registerTemporary(func)
session.udf.registerTemporary(udfName, func)
functions.udf(func)
session.udf.registerPermanent(udfName2, func, stageName)
```
I included all of them in the tests

**Review by sfc-gh-yuwang — APPROVED** [review]
